### PR TITLE
Add markdown-escape

### PR DESCRIPTION
**Pull request by LinqLover**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by LinqLover** [comment]
> 🔔 @LinqLover — you're the only owner, but it would still be good if you find someone to [review this PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56878/files) in the next few days, otherwise a maintainer will look at it. (And if you do find someone, maybe even recruit them to be a second owner to make future changes easier...)

@kemitchell Are you interested? :)

File: types/markdown-escape/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "no-duplicate-imports": ["allow-namespace-imports"]
```
Comment (elibarzilay) [inline types/markdown-escape/tslint.json:4]: Why is this needed?

**Comment by kemitchell** [comment]
@LinqLover I'm not interested. But you do you.

Unsubscribing.

File: types/markdown-escape/tslint.json (hunk 1 above)
Comment (LinqLover) [reply types/markdown-escape/tslint.json:4]: See https://github.com/palantir/tslint/issues/3418. Linter told me to merge these two lines which is not possible:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7e7194659ab906ddebf7168bfb3a66563052af05/types/markdown-escape/markdown-escape-tests.ts#L1-L2

File: types/markdown-escape/markdown-escape-tests.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import * as markdownEscape from 'markdown-escape';
+import { CharacterSet } from 'markdown-escape';
```
Comment (peterblazejewicz) [inline types/markdown-escape/markdown-escape-tests.ts:2]:
```suggestion
import markdownEscape = require('markdown-escape');
import { CharacterSet } from 'markdown-escape';
```
this will remove requirement to add excusion for TSLint
You would write this otherwise:
```ts
import * as markdownEscape from 'markdown-escape';
...
...
markdownEscape.CharacterSet;
```

File: types/markdown-escape/tslint.json (hunk 1 above)
Comment (LinqLover) [reply types/markdown-escape/tslint.json]: Resolved via c812d17557d09a0984988e6ac60b3a2b9fa63800

File: types/markdown-escape/markdown-escape-tests.ts (hunk 2 above)
Comment (LinqLover) [reply types/markdown-escape/markdown-escape-tests.ts]: Resolved via c812d17557d09a0984988e6ac60b3a2b9fa63800

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@LinqLover thx!

**Comment by LinqLover** [comment]
ready to merge

**Comment by LinqLover** [comment]
Will this still be accepted for hacktoberfest? :)
